### PR TITLE
Wrote tests for user post endpoint

### LINKED ISSUE #46: Test POST '/user/'
**Is your feature request related to a problem? Please describe.**
We should write tests for our API endpoints. Great way to catch any bugs, especially those concerning edge cases.

**Describe the solution you'd like**

Write some tests for `POST /user/`. You can write this in `api/test/routes/user.test.js`. The file currently contains a commented-out test that tests creating a valid user with only a user's auth id. The data relating to the user is found in `api/test/mock-data/user-mock-data.js`.

Some things you can test:
- Make sure that the proper HTTP response status code is returned: ` expect(res.status).toBe(200);`
- Make sure that the proper data is returned. If the returned data is a single JSON object, you can create an expected return object in `api/test/mock-data/user-mock-data.js`. If the returned data is an array, you can check the length of the array.
- Make sure that the proper message is returned (not as important as the two tests above).

Also, make sure to consider edge cases like:
- The data sent to the endpoint is missing some fields.
- The data sent to the endpoint has additional fields.

**Additional resources**
- https://jestjs.io/

### PR DESCRIPTION
**Pull request by jessicasingh7**
## Description: 

Wrote tests for user post endpoint

Fixes #46

### CODE REVIEW

File: api/test/mock-data/user-mock-data.js [hunk 1]
```diff
@@ -1,11 +1,49 @@
...
   role: 0,
-  authID: 'ba32cb26-2020-4fbc-b77d-34ea6b0790a6',
+  authID: 'ba32cb26-2020-4fbc-b77d-34ea6b0790a7',
+  adminLanguages: [],
+  learnerLanguages: [],
+  collaboratorLanguages: [],
+};
+
+module.exports.POST_WRONG_USER_NO_AUTH_ID = {
+  role: 0,
```
Comment (ashayp22) [inline api/test/mock-data/user-mock-data.js:15]: use enum instead of hard coding  (multiple places)

File: api/test/mock-data/user-mock-data.js [hunk 2]
```diff
@@ -1,11 +1,49 @@
...
+
+module.exports.POST_WRONG_USER_NO_ROLE = {
+  authID: 'ba32cb26-2020-4fbc-b77d-34ea6b0790a7',
+  adminLanguages: [],
+  learnerLanguages: [],
+  collaboratorLanguages: [],
+};
+
+module.exports.POST_USER_ONE_LESS_FIELD = {
+  authID: 'ba32cb26-2020-4fbc-b77d-34ea6b0790a7',
```
Comment (ashayp22) [inline api/test/mock-data/user-mock-data.js:46]: add role field, remove non-required field

File: api/src/api/user.js [hunk 3]
```diff
@@ -17,6 +17,10 @@ router.post(
   '/',
   errorWrap(async (req, res) => {
     const userInfo = req.body;
+    const exists = await models.User.exists({ authID: userInfo.authID });
+    if (exists) {
+      return sendResponse(res, 202, 'User with this authID already exists');
```
Comment (ashayp22) [inline api/src/api/user.js:22]: return all user data except auth id (also in main section of post request)

**Review by ashayp22 — CHANGES_REQUESTED** [review]
mind blowing

**Review by ashayp22 — APPROVED** [review]
spectacular 🤩
